### PR TITLE
add transfer vpc_endpoint + nlb + transfer service + rename bucket

### PR DESCRIPTION
**Pull request by marwinbaumannsbp**
* I have renamed the buckets in this commit as well since we have a backup of the data in the RAW bucket, and a backup on my desktop. Therefore I think this is the right time to remove 'bucket' from all bucket names.

* I have decided to not use the vpc_endpoint option of the community vpc module for now, since we are switching back to the MCAF vpc module soon and this functionality is not included there yet. 

* the resource `aws_vpc_endpoint.transfer` does not expose the private ip address (https://www.terraform.io/docs/providers/aws/r/vpc_endpoint.html) but only the network_interface_ids. But `aws_lb_target_group_attachment.nlb_transfer_target_groups_attachment.target_id` only accepts an IP address. Therefore I have added the resource `data aws_network_interface` to get the IP addresses, but since this data is not known when first executing this code you get an error (invalid count argument). Because the aws_vpc_endpoint needs to be created first. This is solvable by adding the -target argument and create the aws_vpc_endpoint first. (this is an issue more people are having: https://stackoverflow.com/questions/57886961/terraform-get-subnet-integration-ips-from-vpc-endpoint-subnets-tab).

### CODE REVIEW

File: modules/datalayer/sftp_bucket.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 locals {
-  sftp_bucket_name    = var.environment == "prod" ? "gmt-sftp-datalayer-bucket" : "gmt-sftp-datalayer-bucket-${var.environment}"
+  sftp_bucket_name    = var.environment == "prod" ? "gmt-sftp-datalayer" : "gmt-sftp-datalayer-${var.environment}"
```
Comment (shoekstra) [inline modules/datalayer/sftp_bucket.tf]: If we're going to consider changing the bucket names, can we consider #19 to standardise them?  I agree that "bucket" shouldn't be in a bucket name.

File: modules/datalayer/transfer.tf [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  count = length(aws_vpc_endpoint.transfer.network_interface_ids)
+  id    = sort(aws_vpc_endpoint.transfer.network_interface_ids)[count.index]
+}
+
+resource "aws_lb_target_group_attachment" "nlb_transfer_target_groups_attachment" {
+  count            = length(aws_vpc_endpoint.transfer.network_interface_ids)
+  target_group_arn = aws_lb_target_group.transfer_nlb_tg.arn
+  target_id        = data.aws_network_interface.vpc_endpoint_transfer_interfaces[count.index].private_ip
+  port             = 22
+  depends_on       = [aws_vpc_endpoint.transfer, aws_lb_target_group.transfer_nlb_tg]
```
Comment (shoekstra) [inline modules/datalayer/transfer.tf]: I don't think this is required, at least `aws_lb_target_group.transfer_nlb_tg` shouldn't be as it's referenced as a value so Terraform should know it's a dependency.

File: modules/datalayer/transfer.tf [hunk 3]
```diff
@@ -0,0 +1,101 @@
+locals {
+  transfer_user = "bis"
+}
+
+resource "aws_security_group" "allow_sftp" {
```
Comment (shoekstra) [inline modules/datalayer/transfer.tf]: `allow_sftp` should be `transfer_service` so the group can be used for the service and not just for SFTP.

File: modules/datalayer/transfer.tf [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  internal           = false
+  load_balancer_type = "network"
+  tags               = module.datalayer.tags
+
+  dynamic "subnet_mapping" {
+    for_each = [for i, eip in aws_eip.eip_transfer_nlb : {
+      subnet_id     = tolist(module.vpc.public_subnets)[i]
+      allocation_id = eip.id
+    }]
+    content {
```
Comment (shoekstra) [inline modules/datalayer/transfer.tf]: Blocks should be separated by blank lines, could you add a blank line above this one please.

**Comment by shoekstra** [comment]
> * I have decided to not use the vpc_endpoint option of the community vpc module for now, since we are switching back to the MCAF vpc module soon and this functionality is not included there yet.

This functionality will be ported to the MCAF module as it's quite nice from a user port of view to just switch a bool to enable it.  I live this up to you if you want to switch to it now or do it later when switching to the MCAF module.

Re resource naming, typically in a module you would name each resource `default` so I feel we should name the resources something consistent where possible, such as `transfer_service`, instead of differing names per resource.

File: modules/datalayer/transfer.tf (hunk 2 above)
Comment (marwinbaumannsbp) [reply modules/datalayer/transfer.tf]: I agree, removed the dependency

**Pull request merged**